ted2zim/scraper.py

```diff
@@ -258,471 +308,217 @@ def dump_data(self):
-        TED/build/{video id}/thumbnail.jpg.
-        Save the image of the speaker in TED/build/{video id}/speaker.jpg.
-        """
-
-        self.load_metadata()
+
+        # Download all the TED talk videos and the meta-data for it.
+        # Save the videos in build_dir/{video id}/video.mp4.
+        # Save the thumbnail for the video in build_dir/{video id}/thumbnail.jpg.
+        # Save the image of the speaker in build_dir/{video id}/speaker.jpg.
+        # load the dumped metadata
+        self.load_meta_from_file()
         for video in self.videos:
+            # set up variables
             video_id = str(video[0]["id"])
             video_title = video[0]["title"]
             video_link = video[0]["video_link"]
             video_speaker = video[0]["speaker_picture"]
             video_thumbnail = video[0]["thumbnail"]
-            video_dir = path.join(self.scraper_dir, video_id)
-            video_file_path = path.join(video_dir, "video.mp4")
-            speaker_path = path.join(video_dir, "speaker.jpg")
-            thumbnail_path = path.join(video_dir, "thumbnail.jpg")
-
-            if not path.exists(video_dir):
-                os.makedirs(video_dir)
-
-            if not path.exists(video_file_path):
-                print("Downloading video... " + video_title.encode("utf-8"))
-                for i in range(5):
-                    while True:
-                        try:
-                            r = utils.download_from_site(video_link)
-                            with open(video_file_path, "wb") as code:
-                                code.write(r.content)
-                        except Exception as e:
-                            raise e
-                            sleep(5)
-                            continue
-                        break
-
+            video_dir = self.build_dir.joinpath(video_id)
+            video_file_path = video_dir.joinpath("video.mp4")
+            speaker_path = video_dir.joinpath("speaker.jpg")
+            thumbnail_path = video_dir.joinpath("thumbnail.jpg")
+
+            # ensure that video directory exists
+            if not video_dir.exists():
+                video_dir.mkdir(parents=True)
+
+            # download video
+            if not video_file_path.exists():
+                logger.debug(f"Downloading {video_title}")
+                try:
+                    save_large_file(video_link, video_file_path)
+                except Exception:
+                    logger.error(f"Could not download {video_file_path}")
             else:
-                print(
-                    "video.mp4 already exist. Skipping video "
-                    + video_title.encode("utf-8")
-                )
+                logger.debug(f"video.mp4 already exists. Skipping video {video_title}")
 
             # download an image of the speaker
-            if not path.exists(speaker_path) and video_speaker != "":
+            if not speaker_path.exists() and video_speaker != "":
```

Comment:
`and video_speaker != "":` should be enough

Reply:
This one checks whether the speaker image already exists or not. If it already exists, it skips downloading. However, a bit of change was required here as this led to a misleading message on the log.
I have changed this to -
```
if not speaker_path.exists():
    if video_speaker == "None" or video_speaker == "":
        logger.debug("Speaker doesn't have an image")
    else:
        logger.debug(f"Downloading Speaker image for {video_title}")
        save_large_file(video_speaker, speaker_path)
else:
    logger.debug(f"speaker.jpg already exists for {video_title}")
```

Reply:
hey sorry my message was wrong, I copy-pasted and forgot to edit. What I meant was that `if not speaker_path.exists() and video_speaker != "":` could be replaced with `if not speaker_path.exists() and video_speaker:` (empty string would return false).
Sorry about that.